### DISCUSSION TITLE
Design refresh

### DISCUSSION THREAD

**Discussion by AshKyd** [web-scrobbler/web-scrobbler#2762] [Ideas]
Hey there, I was going to submit a pull request to improve UX in the edit form.

**Before:**
![image](https://user-images.githubusercontent.com/49600/110414152-1301a480-80db-11eb-99be-31d7aa83641e.png)

**After:**
![image](https://user-images.githubusercontent.com/49600/110414088-f1a0b880-80da-11eb-83d6-b3fc8c39d3cb.png)

So I
1. Added the localised text from the notification to explain what the form is for.
2. Put the label outside the field, so you can tell what the field is when it's already got text in it
3. Made the buttons more buttony - I don't know why, but these were text links in dev and icons in prod, which confused me. I think I will revert this.
4. Added some more keyboard accessibility so you don't need your mouse :)

I guess I'm wondering how much change you're willing to accept in a pull request :laughing: and if you have any other design ideas you want me to try to work on.

**Comment by alexesprit** [comment]
1. I like the idea, this label helps to understand what is this input form for.
2. This is also useful, though it increases the popup height (I tried to avoid this).
3. I made these buttons as icons intentionally (see #2330 for details), and I would keep them as is.
4. I guess this could be achieved with the current design, but it's a nice addition.

> though it increases the popup height (I tried to avoid this)

If you increases the popup height, you'll have a free white space, which looks weird and unnatural. Probably, we could hide album art in edit mode, add some height transitions etc.

![image](https://user-images.githubusercontent.com/1119267/110426375-0b300900-80b7-11eb-94de-92e23885649d.png)

About PRs: I made a lot of internal changes in `feature/v3` branch. If you want to make significant changes, it's better to use this branch as a base.

**Reply by AshKyd** [reply]
Ah, I was on an old branch, that was why! I agree with all your points. Will check out the other branch and have a look.

**Reply by alexesprit** [reply]
I found your suggestion a quite interesting for me, so I thought about it in more detail.

I made a quick preview with graphics editor to show my idea to get rid of useless and weird white space. Also, I think we don't really need cover arts in edit mode (they aren't related to edit mode since we cannot use custom cover arts).

<details>
  <summary><b>Preview (w/o cover art)</b> (Click to expand)</summary>

![image](https://user-images.githubusercontent.com/1119267/110506075-d43a1180-810f-11eb-9b52-64e3517736f8.png)
</details>

I was also thinking about regular buttons vs icons, and I think we could use buttons:
1. Since the cover arts are removed, we have more horizontal space for long button labels.
2. Again, if we remove cover arts, edit vs info mode becomes different (I mean they look different), so we would apply different style for them. Moreover, we can change links to buttons in other popups to keep the consistent UI style (see an example below).

<details>
  <summary><b>Example of another popup</b> (Click to expand)</summary>

![image](https://user-images.githubusercontent.com/1119267/110507682-49f2ad00-8111-11eb-991f-64f3ea9e6ba7.png)
</details>

Also we could update UI to keep the same style as we use in modal windows (including the bottom footer for buttons):
<details>
  <summary><b>Example of modal window</b> (Click to expand)</summary>

![image](https://user-images.githubusercontent.com/1119267/110508392-03518280-8112-11eb-9202-9e3d3d41d79b.png)
</details>

On the other hand we use text links everywhere except in modal windows (e.g. in the Accounts tab). Is it ok to use links as buttons? Probably it's okay (buttons for input forms, links for everything else).

<details>
  <summary><b>Screenshot of accounts view</b> (Click to expand)</summary>

![image](https://user-images.githubusercontent.com/1119267/110507843-78708800-8111-11eb-8401-13d0552f5f7b.png)
</details>

These are just my thoughts, I'm not a designer. :) What's your opinion?

**Reply by alexesprit** [reply]
One small note: the edit mode is not for unrecognized tracks only. One can edit any track, even if it was recognized by the extension, and we should use different labels for different cases. There's a way to distinguish unrecognized/recognized tracks programmatically, so it shouldn't be a big issue.
